File: solvers.julia/wpnp/test_num_wpnp.jl
include("../_common/header.jl")
include("inidata_num_wpnp.jl")
include("coefs_wpnp.jl")
include("red_85x107_wpnp.jl")

function test_num_wpnp()

    Random.seed!(23)
    N = 10000
    d = 16

    stats = ini_stats("wpnp")

    for i in 1:N
        data = inidata_num_wpnp()
        C,U = coefs_wpnp(data)

        toc = @elapsed S = red_85x107_wpnp(C)
        stats.tm = [stats.tm; toc]
        if isempty(S); continue; end

        stats = update_stats(stats,S,C,U,d)
    end

    disp_stats(stats)

end

test_num_wpnp()

File: solvers.python/py_wpnp/test_num_wpnp.py
from numpy.random import seed
from time import time
from _common import ini_stats, update_stats, disp_stats
from py_wpnp import inidata_num_wpnp, coefs_wpnp, red_85x107_wpnp

def test_num_wpnp():
    seed(23)
    N = 1000
    d = 16

    stats = ini_stats('wpnp')

    for i in range(N):
        data = inidata_num_wpnp()
        C,U = coefs_wpnp(data)

        tic = time()
        S = red_85x107_wpnp(C)
        stats['tm'].append(time()-tic)
        if S is None:
            continue

        stats = update_stats(stats,S,C,U,d)

    disp_stats(stats)